migrations/helpers/update_filtered_positives_helper.py

```diff
@@ -218,17 +218,17 @@ def update_dart_fields(config: Config, samples: List[SampleDoc]) -> bool:
         cursor = sql_server_connection.cursor()
 
         for plate_barcode, samples_in_plate in groupby_transform(
-            samples, lambda x: x[FIELD_PLATE_BARCODE], reducefunc=lambda x: list(x)
+            samples, lambda x: x[FIELD_PLATE_BARCODE], None, lambda x: list(x)
         ):
             try:
-                labware_class = labclass_by_centre_name[samples_in_plate[0][FIELD_SOURCE]]
+                labware_class = labclass_by_centre_name[(str)(samples_in_plate[0][FIELD_SOURCE])]
```

Comment:
field source is the lab where is from, is it not a string?

Reply:
`samples_in_plate[0]` gives back a SampleDoc which is an alias for a dictionary where the keys are strings and the values are ModifiedRowValue which is an alias for `Optional[Union[str, datetime, bool, int, Decimal128, ObjectId]]`

So the type checking is complaining that `labclass_by_center_name` can only accept a string.  Understandable!  We know that the `FIELD_SOURCE` is a string so I'm just casting it to that type.

Reply:
:thumbsup: